tasks/rh-sign-image/rh-sign-image.yaml

```diff
@@ -116,26 +109,29 @@ spec:
 
             for manifest_digest in $manifest_digests; do
               for tag in ${TAGS}; do
-                for registry_reference in $(jq -r '.[] | .url' <<< $referenceJson); do
-                  echo "Creating InternalRequest to sign image with tag ${tag}:"
-                  echo "- reference=${registry_reference}:${tag}"
-                  echo "- manifest_digest=${manifest_digest}"
-                  echo "- requester=$(params.requester)"
-
-                  internal-request -r "${request}" \
-                      -p pipeline_image=${pipeline_image} \
-                      -p reference=${registry_reference}:${tag} \
-                      -p manifest_digest=${manifest_digest} \
-                      -p requester=$(params.requester) \
-                      -p config_map_name=${config_map_name} \
-                      -l ${TASK_LABEL}=${TASK_ID} \
-                      -l ${PIPELINERUN_LABEL}=$(params.pipelineRunUid) \
-                      -s false
-                  ((++count))
-
-                  if [ "$count" -eq "$N" ]; then
-                      wait-for-ir -l ${TASK_LABEL}=${TASK_ID} -t $(params.requestTimeout)
-                      count=0
+                # Iterate over both rh-registry-repo and registry-access-repo
+                for registry_reference in ${rh_registry_repo} ${registry_access_repo}; do
+                  if [ -n "$registry_reference" ]; then
```

Comment:
I don't think we need the `if` (same for the other block). The value will never be `""`. The way the `for` is constructed, it would just skip empty values. But also, given how we read the repository fields with `jq`, if that field doens't exist, you would end up with "null". We should either just assume that it's correctly defined (because this task is only used in the rh specific pipelines) or maybe we should verify that the fields are defined - if not, fail. 

Ideally we would validate this in apply-mapping - if it's a rh-specific pipeline, make sure registry.redhat.io repo reference is used. But maybe we can add that later (it would require more work).

Reply:
Do you want me to add any validation or suggesting that this can be improved in apply-mapping later as an enhancement?

Reply:
For now I would suggest to just remove these `if` checks, because that should never happen. But for an enhancement, I created this Jira for later: https://issues.redhat.com/browse/RELEASE-1115

Reply:
Thanks, I will do that.

Reply:
fixed!

Reply:
Cool!